### PR TITLE
Consolidate request properties in dev and publish env #111

### LINKED ISSUE #111: Consolidate request properties in dev and publish env
There's currently a mismatch of request properties passed into the `payload` between dev and publish environments.

### PR DESCRIPTION
**Pull request by rofe**
Fixes #111

### CODE REVIEW

File: src/HelixServer.js [hunk 1]
```diff
@@ -86,24 +86,23 @@ function executeTemplate(ctx) {
...
     __ow_method: ctx.method.toLowerCase(),
     __ow_logger: ctx.logger,
     owner: ctx.config.contentRepo.owner,
     repo: ctx.config.contentRepo.repo,
     ref: ctx.config.contentRepo.ref || 'master',
     path: `${ctx.resourcePath}.md`,
+    selector: ctx._selector,
+    extension: ctx._extension,
+    url: ctx._url,
+    params: querystring.stringify(ctx._params),
```
Comment (kptdobe) [inline src/HelixServer.js]: The change implies that now `_params` is never `undefined`, old code was testing for it. I would assume there is a test for it which needs to be adjusted then...

Comment (rofe) [reply src/HelixServer.js]: `querystring.stringify` will return an empty string if `ctx._params` is undefined or an empty object, which is what we test for in [expected_dump.json](https://github.com/adobe/helix-simulator/blob/bf5dd52573b94f80ab907f5b88e9fa622ad3b6c4/test/specs/expected_dump.json#L15)

File: src/HelixServer.js [hunk 2]
```diff
@@ -86,24 +86,23 @@ function executeTemplate(ctx) {
...
     __ow_headers: owHeaders,
     __ow_method: ctx.method.toLowerCase(),
     __ow_logger: ctx.logger,
     owner: ctx.config.contentRepo.owner,
     repo: ctx.config.contentRepo.repo,
     ref: ctx.config.contentRepo.ref || 'master',
     path: `${ctx.resourcePath}.md`,
+    selector: ctx._selector,
+    extension: ctx._extension,
+    url: ctx._url,
```
Comment (tripodsan) [inline src/HelixServer.js]: please note that url is not sent by fastly in VCL. so I wouldn't include it here

Comment (rofe) [reply src/HelixServer.js]: Ok, but I guess we should also remove it from [request.schema.json](https://github.com/adobe/helix-pipeline/blob/master/docs/request.schema.json) then...

@trieloff

Comment (tripodsan) [reply src/HelixServer.js]: see: https://github.com/adobe/helix-cli/blob/master/layouts/fastly/helix.vcl#L317

Comment (tripodsan) [reply src/HelixServer.js]: I would generate the URL request property in the pipeline (in openwhisk.js)

Comment (rofe) [reply src/HelixServer.js]: Or add it in `helix.vcl` :)

Should we add `strain` here (and in the pipeline) instead? Would it be useful for developers?

**Comment by rofe** [comment]
Investigating drop in code coverage...

File: src/HelixServer.js (hunk 2 above)
Comment (rofe) [reply src/HelixServer.js]: > I would generate the URL request property in the pipeline (in openwhisk.js)

ok

Comment (tripodsan) [reply src/HelixServer.js]: > Should we add strain here (and in the pipeline) instead? Would it be useful for developers?

we don't use strains yet in simulator. you can add `strain=default` for completness.

Comment (tripodsan) [reply src/HelixServer.js]: > Or add it in helix.vcl :)

the URL is easily reconstructable with `path` and `param`. no need to add it twice.

Comment (rofe) [reply src/HelixServer.js]: yes, and selector and extension

Comment (rofe) [reply src/HelixServer.js]: > we don't use strains yet in simulator. you can add `strain=default` for completness.

`strain` isn't mentioned in `request.schema.json`, so not sure if we really need it

**Pull request merged**